PYTHON_CODING.md

```diff
@@ -0,0 +1,216 @@
+# Python coding best practices
+
+You are writing code that you and other people will read and modify multiple times, so there are few things to keep in mind that will simplify your life:
+- **Keep it simple, clean and documented.** Your code and your documentation are your best methods section! Write it so that your colleagues can understand it easily, and that you will remember what you did 6 months from now.
+- **Make it robust and extensible.** Writing robust code means you will spend less time debugging, or even worse, analyzing results from an incorrect code. Writing extensible code means you will spend less time modifying it.
+
+This article tries to offer practical suggestions on how to achieve these two (sometimes clashing) goals.
+
+1. [Use a modern environment](#use-a-modern-environment)
+2. [Keep clean and adhere to PEP8](#keep-clean-and-adhere-to-pep8)
+3. [Document your code](#document-your-code)
+4. [Find bugs before they find you](#find-bugs-before-they-find-you)
+5. [Know the basics of OOP design](#know-the-basics-of-oop-design)
+
+
+## Use a modern environment
+
+- **Support Python 3!** Almost all scientific packages have now added Python 3 support, and many are making plans to drop Python 2.
+
+- **Use a good editor.** Until you try, you have no idea how much it can speed you up. It will catch syntactic (and some semantic) errors for you. Autocompletion will save you from misspelling variable/function names. The "jump to definition" (and documentation) feature will spare you a lot of file browsing looking for other code. My personal favorite is PyCharm, but any editor with those features will do.
```

Comment:
Will add the links to these, thanks. I don't have other editors I can personally suggest.